### ISSUE TITLE
Need to figure out what to do with multiple Notations, Provenances, Centuries - whether or not to nest them, etc.

### ISSUE THREAD

**Issue by jacobdgm** [DDMAL/CantusDB#360]
CURRENT STATUS: the three detail pages in question have been implemented. This issue remains open because there are questions of how best to handle sub-provenances, sub-notations, etc - i.e. see the discussion below. /CURRENT STATUS

Currently, on source-detail pages, there are hyperlinks following "Provenance:" and "Date:" in the right sidebar. These hyperlinks currently link to "#", as we don't have any provenance or century views. These views should list all sources connected with that particular provenance/century.

EDIT: we also have a similar dead link following "Notation:".

We should have Detail views for both Provenance and Century, at least. It would not be hard to set up list views for these models, though it would be worth discussing how users should be able to access these list views, if they're implemented (the page header is full, and no other logical place to add a link is coming immediately to mind).

This is mentioned in the "Wishlist" google doc (referring, it seems, to behavior currently exhibited on OldCantus):
> hyperlinks to some controlled vocabularies

> "these should all work
> For instance, if I click ""14th century"" in a list of sources, it doesn't pull up everything from the 14th century like I think it should (it's blank).
> Also, if I click on a provenance place name, I want to get some results of sources from there (and it's also blank)."

**Comment by annamorphism** [comment]
is this GoogleDoc wishlist viewable somewhere?
For this to work nicely, it seems like you'd want the centuries/provenances to be sort of nested, e.g  "14th century, 2nd half" should turn up along with "14th century"--as they already do in the search results for "14th century". (Which I guess raises the question of whether there is a point to a list view that basically replicates the search results.)
Geography would be trickier--I don't think a list view of the single source tagged "France" is useful to anybody, really, but unless somebody goes through all the sources and adds other data to them there's not much else to get. This needs some thought.

**Comment by jacobdgm** [comment]
> is this GoogleDoc wishlist viewable somewhere?

You should have access now!

Yeah, glancing at the Centuries currently in the database, some seem to represent full centuries, some half centuries, and some quarter centuries. There may be an easy way to set up a nesting structure (though someone would have to go through the data and set up the nesting relationship by hand, I think. There are only ~50 of them, though, so it might not be too onerous).

If I was in charge of the data in CantusDB, judging by what's already in there, I'd just break things down into quarter-centuries, and assign manuscripts to multiple "Centuries". But that's not my job, so I'll let the musicologists sort out how you all want things to be saved.

**Comment by jacobdgm** [comment]
I added an edit to the initial post - there's a similar dead link for Notation objects.

@annamorphism , is there nesting within Notations as well?

**Comment by jacobdgm** [comment]
One possibility is to set up simple detail views for the Centuries/Provenances/Notations we currently have, and eventually come back, adding `sub_centuries` and `super_century` (or `child_centuries`/`parent_century`) fields (and analogous ones for provenances/notations), and then change the detail views to display the `sub_` sources as well.

The "eventually come back" step would definitely go into "enhancement" territory, and would involve adding data to the database manually to make it work. We should check with Debra etc. before embarking on this, to make sure everyone actually wants it.

**Comment by annamorphism** [comment]
@jacobdgm yes, I think Notations should also be sort of nested. The current notations field (only applied to Sequences) has some strange options, but show this nested idea (Rheinmass:square vs Rheinmass: Lotharingian). Or maybe there should be several facets related at the sibling level (which, again, would mean coming back and adding/changing data.)
